feat: Allow remote editor for WPCOM Simple sites

dcalhoun opened the pull request:
Allow WPCOM Simple sites to use the experimental remote, site-specific editor. Close CMM-223.

Related: 

- CMM-223-linear-issue
- 177481-ghe-Automattic/wpcom
- https://github.com/Automattic/jetpack/pull/42314
- https://github.com/wordpress-mobile/GutenbergKit/pull/109
- https://github.com/wordpress-mobile/WordPress-iOS/pull/24170

## To Test:

<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->

1. Navigate to Me → Experimental Features and enable `Experimental block editor plugins`
1. Open a post on a WPCOM Simple site. 
1. Insert non-core Jetpacks blocks. 
1. Verify they function as expected. 

## Regression Notes

1. Potential unintended areas of impact
    The local, bundled experimental editor. 
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    Manually smoke tested. 
3. What automated tests I added (or what prevented me from doing so)
    Deemed unnecessary for this experimental feature. 

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

dcalhoun committed:
feat: Expose experimental editor plugins feature

dcalhoun committed:
feat: Configure the editor using the plugins feature flag

Allow toggling plugin support within the editor.

dcalhoun committed:
feat: Pass all siteApiNamespace options

The editor needs both namespace patterns to accurately located their
existence in network requests.

dcalhoun committed:
feat: Disable namespacing for select paths

Some endpoints are global, not site specific.

dcalhoun committed:
refactor: Utilize GutenbergKit's EditorConfiguration class

dcalhoun committed:
feat: Set `_currentSiteType` global required by `jetpack-connection`

Without this global, Jetpack blocks perceive a lack of an account
connection.

dcalhoun committed:
feat: Feature config controls experimental editor plugin availability

Allow disabling the plugin support as needed--e.g., while awaiting the
REST API endpoint to deploy or due to a breakage.

dcalhoun committed:
build: Update GutenbergKit ref

dangermattic commented:
<!--
  0 Errors
  2 Warnings: <code>strings.xml</code> files...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
feat: Limit GutenbergKit plugin support to Simple sites

Application password authentication is required for
non-Wordpress.com-hosted sites, as certain blocks attempt to make API
requests to the site domain.

dcalhoun commented on WordPress/src/main/java/org/wordpress/android/util/config/GutenbergKitPluginsFeatureConfig.kt:
Used to ensure we can remotely control whether the plugin feature is available—for API endpoint rollout or breakages.

dcalhoun replied:
This remote feature flag was removed in https://github.com/wordpress-mobile/WordPress-Android/pull/21773/commits/f346ccc5016ec8943ee59fca92b70d0c479a54c7 based on feedback in https://github.com/wordpress-mobile/WordPress-iOS/pull/24170#issuecomment-2769617420 that is may be unnecessary. The experimental feature toggle remains in place.

dcalhoun committed:
style: Address line length lint warning

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
refactor: Update remote feature flag key

Match the latest server implementation.

dcalhoun committed:
feat: Remove remote feature flag

This controlled the availability of the experimental feature toggle and
also could override the feature flag status. This additional remote
feature flag was deemed unnecessary for an experimental feature.

https://github.com/wordpress-mobile/WordPress-iOS/pull/24170#issuecomment-2769617420

dcalhoun commented:
FYI: the related server changes adding the REST API endpoint were deployed, so there is no longer a need to route device traffic to a sandbox server. I updated the PR description accordingly.

jkmassel reviewed (approved):
I tested this using the "Business Hours" and "AI Assistant" blocks.

Both worked great.

dcalhoun committed:
build: Update GutenbergKit ref
